lib/HLSL/DxilValidation.cpp

```diff
@@ -2345,7 +2372,65 @@ static void ValidateDxilOperationCallInProfile(CallInst *CI,
           CI, ValidationRule::SmOpcodeInInvalidFunction,
           {"CreateHandle", "Shader model 6.5 and below"});
     }
-    break;
+  } break;
+      
+  case DXIL::OpCode::ThreadId: { // SV_DispatchThreadID
+    if (shaderKind != DXIL::ShaderKind::Node) {
+      break;
+    }
+
+    if (NLT == DXIL::NodeLaunchType::Broadcasting) {
+      // good
+    } else {
```

Comment:
It's ok, but why this pattern instead of `!= DXIL::NodeLaunchType::Broadcasting`?

Reply:
I found it easier to read to do the pattern shown above. If I instead specified the failure case logically, we would need to write something like ` ! (NLT == coalescing || NLT == thread)`, just a tad bit more complicated.

Reply:
I also found this pattern confusing. Maybe it would be clearest if you use break instead of an empty if block?
```c++
    if (nodeLaunchType == DXIL::NodeLaunchType::Broadcasting)
      break;
    ValCtx.EmitInstrFormatError(...);
```